### ISSUE TITLE
Microsoft.Azure.Storage.Common dependency

### ISSUE THREAD

**Issue by seniorquico** [Azure/azure-cosmos-table-dotnet#3]
TLDR: Is it planned and/or possible to refactor this library to depend on the common classes found in `Microsoft.Azure.Storage.Common` 10.x? I'm trying to migrate code from `WindowsAzure.Storage` 9.3.3, but it's proving *very* difficult due to the duplication of common classes (`StorageException`, `CloudStorageAccount`, etc.).

----

We have a system built with Microsoft Orleans on .NET Standard, and we're using the built-in storage and stream providers for Azure Storage-based services (Blob, Queue, and Table). An issue keeps cropping up with `WindowsAzure.Storage` 9.3.3:

https://github.com/Azure/azure-storage-net/issues/620

A contributor submitted a patch to try and fix the issue, but it was closed since Table has transitioned to the Cosmos DB team:

https://github.com/Azure/azure-storage-net/pull/774

Sorry for the quick rant... but this transition has been rough... 7 months later, and we're still waiting for serious progress on the .NET Standard version of this package:

https://github.com/Azure/azure-storage-net/issues/779

I tried to take a stab at "upgrading" the Microsoft Orleans providers to the newer libraries (`Microsoft.Azure.Storage.Blob`, `Microsoft.Azure.Storage.Queue`, and `Microsoft.Azure.Cosmos.Table`):

https://github.com/dotnet/orleans/issues/5479

As noted in that PR, this is a difficult transition due to utilities like [`AzureStorageUtils`](https://github.com/dotnet/orleans/blob/v2.3.2/src/Azure/Shared/Storage/AzureStorageUtils.cs) that try to take care of cross-cutting concerns, like parsing details from `StorageException`. Additionally, some classes/methods have had their visibility changed.

Based on my research, the new Azure Storage SDKs have moved common classes like `StorageException` into `Microsoft.Azure.Storage.Common`. So back to the question, is it planned and/or possible to refactor this library to depend on `Microsoft.Azure.Storage.Common`? Even if the answer is "no", I think that would help projects, like Microsoft Orleans, come up with a migration strategy.

**Comment by donghexu** [comment]
Hi @seniorquico, we the Cosmos DB team has developed a new .net standard sdk Microsoft.Azure.Cosmos.Table (https://www.nuget.org/packages/Microsoft.Azure.Cosmos.Table), which removed the dependency on Storage.Common and remained classes and advanced features. Please use that to unblock you. Here is the sample project for reference: https://github.com/Azure-Samples/azure-cosmos-table-dotnet-core-getting-started. 

Thank you,

**Comment by seniorquico** [comment]
@donghexu Yes, I know that package exists. I referenced it in the original post. It's the _lack_ of the dependency on `Microsoft.Azure.Storage.Common` that's blocking our migration. Are you confirming that Microsoft Orleans will need to duplicate all of the common code, e.g. parsing `StorageException`, to support the new `Microsoft.Azure.Cosmos.Table` SDK?

**Comment by vmendi** [comment]
Hi!

This issue is also affecting us and we'd love to see Microsoft Orleans ported to the new Microsoft.Azure.Cosmos.Table as soon as possible. 

PR's to the old table library are being closed because of this transition, and those fixes were important!

Please let us know if we can do anything to help.

**Comment by wmengmsft** [comment]
Hi @seniorquico, the Cosmos DB Table SDK is an entirely new Table SDK designed to cater for the needs of both Azure Storage Table API and Cosmos DB Table API. We retained most of the API surface from Azure Storage to reduce the learning curve / transition cost. However, when it comes to cross-cutting APIs such as OperationContext and StorageException, we find the need to fork the implementation due to diverging feature-sets and requirements. Hence, at this time we don't have plans to move Cosmos.Table SDK to depend on Storage.Common.

If needed, we can help with any specific issues Microsoft Orleans encounter when migrating and provide recommendations.

**Comment by seniorquico** [comment]
Thank you for confirming the direction of this library, @wmengmsft. I'll relay this information back to the Orleans issue.

**Comment by seniorquico** [comment]
@wmengmsft Sorry to double post, but...

> We retained most of the API surface from Azure Storage to reduce the learning curve / **transition cost**.

Transition cost? Sorry, I don't see how this new library helps reduce it. I have a large number of use cases where blob & table code are intermixed, especially for cross-cutting concerns. Microsoft Orleans was really just one example. Given the fork, I'm now looking at a very significant effort to migrate.
